Feat: service and permission layers

yld-weng opened the pull request:
- Adding service and permission layers for maintaining separation of concerns.

- Fix the login url, where previously redirected to `/accounts/login` which doesn't exists.

- Added breadcrumb to organisation, projects, and survey pages for better navigation.

![image](https://github.com/user-attachments/assets/a5e43473-9277-4fc2-8ccd-a1913034d36b)

![image](https://github.com/user-attachments/assets/2dfee021-8e06-4d2d-9e47-cf505c6d9478)

![image](https://github.com/user-attachments/assets/cb948132-985c-4739-a8ed-d5a3e7a29f6c)

- Added test fixtures to speed up set up.
- Added a new model `GuestProjectAccess` for restricting guest users.

twinkarma commented:
@yld-weng I don't think we will have guests in the system, only project managers and maybe a sys admin/superuser. Participants (people who fills in the survey) should never need to login and will instead be provided tokens.

yld-weng commented:
> @yld-weng I don't think we will have guests in the system, only project managers and maybe a sys admin/superuser. Participants (people who fills in the survey) should never need to login and will instead be provided tokens.

Thanks for the clarification. I've removed the `Guest` role and renamed `member` to `project manager`.

twinkarma reviewed (approved):
LGTM, I'd like a discussion about permissions but this can be done in a separate issue

twinkarma commented:
@yld-weng is this ready to be merged?

f-allian commented on data/001_users.json:
```diff
@@ -0,0 +1,40 @@
+      "email": "admin@test.com",
+      "first_name": "Admin",
+      "last_name": "User",
+      "password": "pbkdf2_sha256$870000$generate something random$jxrzQIBcrwqeZdLY5hJxM5tugSuSyGIDAsVVx2qIWBY=",
+      "is_staff": true,
```
@yld-weng I've read your README on this, so what are these `password` keys here?

f-allian commented on home/services/base.py:
@yld-weng I've never seen this `services/` folder before. What do the files in here do exactly?

yld-weng replied:
The `services/` folder is where we keep service files that handle the business logic for the Django app. These files decouple business logic from the views and models, making the code more modular and reusable  - especially we have seen similar logic (like permission checks) needs to be applied across multiple views. Each service file typically contains functions or classes that perform specific tasks, in this case, they are for managing projects and organisations.

This service layer pattern, while not part of Django's official documentation, draws inspiration from, I believe the clean architecture book and was seen in many other frameworks.

yld-weng replied:
This pattern was also recommended in [this book](https://www.amazon.co.uk/Ultimate-Django-Development-Using-Python/dp/8196815115) I'm currently reading.
